src/dbnode/persist/fs/commitlog/chunk_reader.go

```diff
@@ -80,18 +83,31 @@ func (r *chunkReader) readHeader() error {
 		return err
 	}
 
-	// Verify data checksum
-	data, err := r.buffer.Peek(int(size))
+	// Setup a chunk data buffer so that chunk data can be loaded into it.
```

Comment:
Good catch on this! A thought: M3DB is generally speaking pretty allocation sensitive (I can't speak to this particular code path, but you'll see all manner of pooling etc to get around allocating new byte slices). 

Since all we're doing here is checksumming, do we actually need to allocate the byte slice? Looks like we may be able to do: